searx/engines/google_images.py

```diff
@@ -156,55 +165,47 @@ def response(resp):
         return results
 
     root = root[0]
-    for img_node in eval_xpath(root, './/img[contains(@class, "rg_i")]'):
-
-        try:
-            img_alt = eval_xpath(img_node, '@alt')[0]
-
-            img_base64_id = eval_xpath(img_node, '@data-iid')
-            if img_base64_id:
-                img_base64_id = img_base64_id[0]
-                thumbnail_src = img_bas64_map[img_base64_id]
+    for img_node in eval_xpath_list(root, './/img[contains(@class, "rg_i")]'):
+
+        img_alt = eval_xpath_getindex(img_node, '@alt', 0)
+
+        img_base64_id = eval_xpath(img_node, '@data-iid')
```

Comment:
Equivalent (?) behavior using `eval_xpath_getindex`:
```python
img_base64_id = eval_xpath_getindex(img_node, '@data-iid', 0, default=None)
if img_base64_id:
    thumbnail_src = img_bas64_map[img_base64_id]
else:
    thumbnail_src = eval_xpath_getindex(img_node, '@src', 0, default='')
    if thumbnail_src == '':
        thumbnail_src = eval_xpath_getindex(img_node, '@data-src', 0, default='')
```

Reply:
Hm, it is equivalent, but what do we win with?

`eval_xpath_getindex` call at first `eval_xpath` and this function will raise `SearxXPathSyntaxException` and `SearxXPathSyntaxException` if something is wrong.

On a empty match, `bool(img_base64_id)` is the same as setting a default when the index 0 does not exists.

If I am not wrong the `eval_xpath_getindex` is equivalent but with additional conditions, checking that the range is OK and the default is not `NOTSET`

https://github.com/searx/searx/blob/0f18e885bf1996603fd6b3021c742b1126abda90/searx/utils.py#L602-L606

Or did I miss something?

Reply:
It is easier for me to read the `eval_xpath_getindex` version, but I wrote that function so, I most probably not objective.

I try to see how far we can enforce checking using different dedicated functions (`eval_xpath_*`, etc...).

The current version is perfectly fine.

Reply:
I think we need a review for each engine, to make them more search.checker conform. This can be done successively when we see a engine-scrapper is broken and needs a fix.